pep-0628.txt

```diff
@@ -9,7 +9,6 @@ Content-Type: text/x-rst
 Created: 28-Jun-2011
 Python-Version: 3.6
 Post-History: 28-Jun-2011
-Resolution: https://bugs.python.org/issue12345
```

Comment:
Looks like this thread does include the resolution, Guido's "I'm just going to do this" at https://bugs.python.org/issue12345#msg272287 ?

Reply:
I see; I'd elided the link from here since it didn't link directly to any resolution message, just a BPO issue, was the one remaining special-case for both the rendering and validation logic for this field, and was already mentioned and linked in the `PEP Acceptance` notice just below it.

In any case, thanks to your feedback, I went ahead and linked `accepted` in that block directly to the message above, while retaining the existing link to the issue itself.

Reply:
Worth noting that BPO will move to GitHub soon (expected next weekend). There will be redirects provided, but not sure they'll work for individual messages.

Reply:
My understanding from following that is that redirects will be available, but they won't be the current URLs, and BPO won't be shut down, it will only become read-only. So this URL _should_ continue to work, like the many others in PEPs. If per-message linking stopped working, that would be a pretty significant breakage. See [@hugovk 's reply](https://discuss.python.org/t/github-issues-migration-is-coming-soon/13791/30?u=cam-gerlach) to my question in the thread, quoting psf/gh-migration#17 ,

However, I understand its not 100% clear from either the [Discourse thread](https://discuss.python.org/t/github-issues-migration-is-coming-soon/13791/28) or the devguide documentation (which took quite a bit of digging in the PR preview to even find). I can ask @ambv to confirm there.

Reply:
Maybe while we wait for GitHub to allow the migration we could update the PEP for the migration with details like which urls will redirect where?

Reply:
If you want to link to the message directly you basically have two options:
* keep the current bpo link, that will continue to work for the foreseeable future (bpo will be around in read-only mode);
* wait until the migration is complete, and replace the bpo link with a link to the corresponding GitHub comment.

Reply:
> keep the current bpo link, that will continue to work for the foreseeable future (bpo will be around in read-only mode)

Given PEPs are historical documents, I don't see a problem with retaining the BPO links for existing, non-Draft/Active PEPs.

On a related note, I was thinking of adding `:bpo:` and `:gh;` roles, based on the current `:issue:` and `:gh:` in the CPython documentation, much like we do for `:pep:` and `:rfc:`. This would make linking to BPO and GitHub issues and PRs easier (just enter the number and it does the rest), while allowing us to make the render nicely and consistently, and most importantly easily update where the links point and how they render without having to touch any individual PEP or require authors to do anything extra. For example, we could (if we wanted/needed) switch `:bpo:` to point to the BPO redirect service to automatically redirect to the corresponding GitHub issue.